### PR TITLE
MBS-10531: Document and fix handling of maps service access token

### PR DESCRIPTION
**Pull request by yvanzo**
# Fix MBS-10531: Invalid requests are sent to maps service when access token is not set

- Document `MAPBOX_ACCESS_TOKEN` in sample config
- Prevent sending unauthorized requests to the maps service
- Display explanatory message instead of incomplete map

### CODE REVIEW

File: root/area/AreaPlaces.js [hunk 1]
```diff
@@ -36,8 +37,19 @@ const AreaPlaces = ({
...
+        {DBDefs.MAPBOX_ACCESS_TOKEN ? (
+          <>
+            <div id="largemap" />
+            {manifest.js('area/places-map.js', {'data-args': mapDataArgs})}
+          </>
+        ) : (
+          <p>
+            {l(
+              `A map cannot be shown because maps service access token
+               is not set for this server.`,
```
Comment (reosarevok) [inline root/area/AreaPlaces.js]: Nitpick, maybe `A map cannot be shown because no maps service access token has been set for this server.` reads better?

Comment (yvanzo) [reply root/area/AreaPlaces.js]: Changed in https://github.com/metabrainz/musicbrainz-server/pull/1334/commits/7219ad43fdc0f3347099cdf972fc2d27fa167745, thanks.

File: lib/DBDefs.pm.sample [hunk 2]
```diff
@@ -374,6 +374,10 @@ sub WEB_SERVER                { "www.musicbrainz.example.com" }
 # sub COVER_ART_ARCHIVE_UPLOAD_PREFIXER { shift; sprintf("//%s.s3.us.archive.org/", shift) };
 # sub COVER_ART_ARCHIVE_DOWNLOAD_PREFIX { "//coverartarchive.org" };
 
+# Mapbox access token mapbox must be set to display area/place maps.
```
Comment (mwiencek) [inline lib/DBDefs.pm.sample]: (dupe word here)
```suggestion
# Mapbox access token must be set to display area/place maps.
```

File: lib/DBDefs/Default.pm [hunk 3]
```diff
@@ -358,6 +358,7 @@ sub COVER_ART_ARCHIVE_SECRET_KEY { };
 sub COVER_ART_ARCHIVE_UPLOAD_PREFIXER { shift; sprintf("//%s.s3.us.archive.org/", shift) };
 sub COVER_ART_ARCHIVE_DOWNLOAD_PREFIX { "//coverartarchive.org" };
 
+# Mapbox access token mapbox must be set to display area/place maps.
```
Comment (mwiencek) [inline lib/DBDefs/Default.pm]:
```suggestion
# Mapbox access token must be set to display area/place maps.
```

File: root/place/PlaceMap.js [hunk 4]
```diff
@@ -30,10 +31,19 @@ const PlaceMap = ({
...
+      DBDefs.MAPBOX_ACCESS_TOKEN ? (
+        <>
+          <div id="largemap" />
+          {manifest.js('place/map.js', {'data-args': mapDataArgs})}
+        </>
+      ) : (
+        <p>
+          {l(
+            `A map cannot be shown because maps service access token is not set
+             for this server.`,
```
[the remark this answers is not in the document]
Comment (yvanzo) [reply root/place/PlaceMap.js]: Oops, fixed with https://github.com/metabrainz/musicbrainz-server/pull/1334/commits/0f3135e83d180ba2bdb34f9b11adf4fa6ac2778e#diff-45c3071fc9307c35c1eb718f698de086R42-R43

**Pull request merged**